common/src/lib.rs

```diff
@@ -267,6 +270,40 @@ where
     } else {
         None
     };
+    let ramdisk_slice_len = system_info.ramdisk_len;
+    let ramdisk_slice_start = if let Some(ramdisk_address) = system_info.ramdisk_addr {
+        let ramdisk_address_start = mapping_addr(
+            config.mappings.ramdisk_memory,
+            system_info.ramdisk_len,
+            Size4KiB::SIZE,
+            &mut used_entries,
+        );
+        let physical_address = PhysAddr::new(ramdisk_address);
+        let ramdisk_physical_start_page: PhysFrame<Size4KiB> =
+            PhysFrame::containing_address(physical_address);
+        let ramdisk_page_count = system_info.ramdisk_len / Size4KiB::SIZE;
```

Comment:
We still need the first `-1` because otherwise we allocate one page more than needed if the size is exactly a multiple of `Size4KiB::SIZE`.

```suggestion
        let ramdisk_page_count = (system_info.ramdisk_len - 1) / Size4KiB::SIZE;
```